src/lib/config.ts

```diff
@@ -137,10 +137,28 @@ export const DB_QUERY_TIMEOUT = intVar(
 export const DELTA_HOST = requiredVar('DELTA_HOST');
 export const FILES_HOST = optionalVar('FILES_HOST', '');
 export const DEVICE_CONFIG_OPENVPN_CA = requiredVar('DEVICE_CONFIG_OPENVPN_CA');
-export const DEVICE_CONFIG_OPENVPN_CONFIG = openVpnConfig.replace(
-	'remote {{VPN_DETAILS}}',
-	`remote ${requiredVar('VPN_HOST')} ${requiredVar('VPN_PORT')}`,
-);
+export const VPN_UDP_HOST = optionalVar('VPN_UDP_HOST');
+export const VPN_HOST = requiredVar('VPN_HOST');
+export const VPN_PORT = requiredVar('VPN_PORT');
+
+const remotes = [];
+let tmpConf = openVpnConfig;
+if (VPN_UDP_HOST) {
+	remotes.push(`remote ${VPN_UDP_HOST} ${VPN_PORT} udp`);
+	remotes.push(`remote ${VPN_HOST} ${VPN_PORT} tcp`);
+	tmpConf = openVpnConfig
+		.replace('{{VPN_DETAILS}}', remotes.join('\n'))
+		.split('\n')
+		.filter((el) => el !== 'proto tcp')
+		.join('\n');
+} else {
+	tmpConf = openVpnConfig.replace(
+		'{{VPN_DETAILS}}',
+		`remote ${VPN_HOST} ${VPN_PORT}`,
+	);
+}
+export const DEVICE_CONFIG_OPENVPN_CONFIG = tmpConf;
```

Comment:
Tiny-NIT: This should be equivalent:
```suggestion
export const DEVICE_CONFIG_OPENVPN_CONFIG = (() => {
	if (VPN_UDP_HOST) {
		const remotes = [
			`remote ${VPN_UDP_HOST} ${VPN_PORT} udp`,
			`remote ${VPN_HOST} ${VPN_PORT} tcp`,
		].join('\n');
		return openVpnConfig
			.replace('{{VPN_DETAILS}}', remotes)
			.replace('\nproto tcp\n', '\n')

	} else {
		return openVpnConfig.replace(
			'{{VPN_DETAILS}}',
			`remote ${VPN_HOST} ${VPN_PORT}`,
		);
	}
})();
```